src/main/java/org/rumbledb/expressions/postfix/DynamicFunctionCallExpression.java

```diff
@@ -30,16 +31,21 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public class DynamicFunctionCallExpression extends PostfixExpression {
+public class DynamicFunctionCallExpression extends Expression {
 
+    private Expression mainExpression;
     private List<Expression> arguments;
 
     public DynamicFunctionCallExpression(
             Expression mainExpression,
             List<Expression> arguments,
             ExceptionMetadata metadata
     ) {
-        super(mainExpression, metadata);
+        super(metadata);
+        if (mainExpression == null) {
+            throw new OurBadException("Main expression cannot be null in a postfix expression.");
+        }
+        this.mainExpression = mainExpression;
         this.arguments = arguments;
         if (this.arguments == null) {
             this.arguments = new ArrayList<>();
```

Comment:
Can you take another look at the initHighestExecutionMode() method of this class. The override is probably redundant now as it doesn't inherit from PostFixExpression. But the explicit definition and comment may be helpful for clarity.

Reply:
Good catch! It is removed.